[DNNL]Upgrade the library from v1.1 to v1.2

lisa0314 opened the pull request:
macOS build using dnnl_mac_1.2.0_cpu_iomp.tgz release
https://github.com/intel/mkl-dnn/releases/download/v1.2/dnnl_mac_1.2.0_cpu_iomp.tgz
Linux build using dnnl_lnx_1.2.0_cpu_iomp.tgz release
https://github.com/intel/mkl-dnn/releases/download/v1.2/dnnl_lnx_1.2.0_cpu_iomp.tgz
windows build using dnnl_win_1.2.0_cpu_vcomp release
https://github.com/intel/mkl-dnn/releases/download/v1.2/dnnl_win_1.2.0_cpu_vcomp.zip

lisa0314 commented:
@sunlin-link Fixes [intel/webml-polyfill#164](https://github.com/intel/webml-polyfill/issues/164)

fujunwei commented on services/ml/execution_impl_mac.mm:
```diff
@@ -480,15 +454,12 @@ bool GetMPSImageInfo(const OperandMac& operand,
             if (!dst_img) {
               if (tmp_mpsimage_cache.find(operation_output_idx) ==
                   tmp_mpsimage_cache.end()) {
-                MPSImageDescriptor* descriptor = CreateMPSImageDescriptor(operation_input);
```
Maybe @sunlin-link need take a look.

huningxin commented:
@lisa0314 , I see many changes related to logging and coding style. Could you please explain the root cause and which part of your PR fixes that? Thanks!

lisa0314 committed:
[DNNL]Upgrade the library from v1.1 to v1.2
macOS build using dnnl_mac_1.2.0_cpu_iomp.tgz release
https://github.com/intel/mkl-dnn/releases/download/v1.2/dnnl_mac_1.2.0_cpu_iomp.tgz
Linux build using dnnl_lnx_1.2.0_cpu_iomp.tgz release
https://github.com/intel/mkl-dnn/releases/download/v1.2/dnnl_lnx_1.2.0_cpu_iomp.tgz
windows build using dnnl_win_1.2.0_cpu_vcomp release
https://github.com/intel/mkl-dnn/releases/download/v1.2/dnnl_win_1.2.0_cpu_vcomp.zip

Christywl commented:
Test results on Windows: 
https://brucedai.github.io/webnnt/test/index-local.html?prefer=fast
passes: 318 failures: 419(baseline: "pass": 316, "fail": 421)
No regression issue for workload testing

huningxin commented:
Ping @lisa0314 , please share the results on macOS and Linux. Thanks.

Christywl commented:
Test results on macOS:
https://brucedai.github.io/webnnt/test/index-local.html?prefer=fast
passes: 318 failures: 419(baseline: "pass": 316, "fail": 421)
No regression issue for workload testing

Christywl commented:
Test results on Linux:
https://brucedai.github.io/webnnt/test/index-local.html?prefer=fast
passes: 318 failures: 419(baseline: "pass": 316, "fail": 421)

lisa0314 commented:
@huningxin , Lei help to share the results for three platforms.  Please help to review. Thanks!

huningxin commented:
Looks good, thanks much @lisa0314 and @Christywl
